hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -425,9 +426,6 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			dest.Data["credentials"] = srcData
 			return nil
 		})
-		if err != nil {
```

Comment:
Why is the error handling being deleted here?

Reply:
Just create a noop switch case for None rather than removing the error handling on the AWS path